cmake/modules/FindLibUV.cmake

```diff
@@ -0,0 +1,138 @@
+# This is adapted from
+# https://github.com/Kitware/CMake/blob/master/Source/Modules/FindLibUV.cmake,
```

Comment:
starting from what cmake version is the standard FindLibUV.cmake available?
Why do we need the LibUV_ROOT addition?

I really really want to avoid needing to maintain this (btw, by adding it you're subscribing to maintain the file).

Reply:
I do not know when FindLibUV.cmake will become official in cmake as I do not find any discussion on this.

For LibUV_ROOT, it is necessary as our distribution tests (specifically this one examples/cpp/helloworld/cmake_externalproject/CMakeLists.txt) do not install libuv under the default path for cmake. The test needs to tell the find module where to search for the libuv installation. This idea is borrowed from the standard [FindZLIB.cmake module](https://github.com/Kitware/CMake/blob/master/Modules/FindZLIB.cmake#L46).

Unfortunately we must keep this find module as libuv does not generate a config.cmake file (thus find_package(libuv CONFIG) does not work). @muxi will take this over in the future as I am going to focus more on grpc development inside Google after this project. (I have asked Muxi to monitor all the changes in this PR).